File: day-1/s3_password-hashing/python-flask/argon2_demo.py
"""Compare two salted Argon2id hashes of one fictional password."""

from __future__ import annotations

from time import perf_counter

from argon2 import PasswordHasher

DEMO_PASSWORD = "password"


def create_argon2_hashes(password: str) -> tuple[str, str, float]:
    hasher = PasswordHasher()
    started_at = perf_counter()
    first_hash = hasher.hash(password)
    second_hash = hasher.hash(password)
    return first_hash, second_hash, perf_counter() - started_at


def main() -> None:
    hasher = PasswordHasher()
    first_hash, second_hash, elapsed_seconds = create_argon2_hashes(DEMO_PASSWORD)
    both_verify = hasher.verify(first_hash, DEMO_PASSWORD) and hasher.verify(
        second_hash, DEMO_PASSWORD
    )

    print("Même mot de passe fictif, deux résultats Argon2id :")
    print(f"1. {first_hash}")
    print(f"2. {second_hash}")
    print(f"Les deux valeurs sont différentes : {first_hash != second_hash}")
    print(f"Les deux vérifications réussissent : {both_verify}")
    print(f"Temps mesuré pour deux calculs sur cette machine : {elapsed_seconds:.3f} s")


if __name__ == "__main__":
    main()

File: day-1/s3_password-hashing/node-express/argon2-demo.js
"use strict";

const { performance } = require("node:perf_hooks");
const argon2 = require("argon2");

const demoPassword = "password";

async function createArgon2Hashes(password) {
  const startedAt = performance.now();
  const firstHash = await argon2.hash(password);
  const secondHash = await argon2.hash(password);
  return { firstHash, secondHash, elapsedMilliseconds: performance.now() - startedAt };
}

async function main() {
  const { firstHash, secondHash, elapsedMilliseconds } = await createArgon2Hashes(demoPassword);
  const bothVerify =
    (await argon2.verify(firstHash, demoPassword)) &&
    (await argon2.verify(secondHash, demoPassword));

  console.log("Même mot de passe fictif, deux résultats Argon2id :");
  console.log(`1. ${firstHash}`);
  console.log(`2. ${secondHash}`);
  console.log(`Les deux valeurs sont différentes : ${firstHash !== secondHash}`);
  console.log(`Les deux vérifications réussissent : ${bothVerify}`);
  console.log(`Temps mesuré pour deux calculs sur cette machine : ${elapsedMilliseconds.toFixed(0)} ms`);
}

if (require.main === module) {
  main().catch((error) => {
    console.error(error.message);
    process.exitCode = 1;
  });
}

module.exports = { createArgon2Hashes };
